arkouda/dataframe.py

```diff
@@ -1841,6 +1844,269 @@ def numeric_help(d):
+        --------
+        unregister, attach, unregister_dataframe_by_name, is_registered
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
+        """
+
+        self.index.register(f"df_index_{user_defined_name}")
+        array(self.columns).register(f"df_columns_{user_defined_name}")
+
+        for col, data in self.data.items():
+            data.register(f"df_data_{col}_{data.objtype}_{user_defined_name}")
+
+        self.name = user_defined_name
+        return self
+
+    def unregister(self):
+        """
+        Unregister this DataFrame object in the arkouda server which was previously
+        registered using register() and/or attached to using attach()
+
+        Raises
+        ------
+        RegistrationError
+            If the object is already unregistered or if there is a server error
+            when attempting to unregister
+
+        See also
+        --------
+        register, attach, unregister_dataframe_by_name, is_registered
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
+        """
+
+        if not self.name:
+            raise RegistrationError(
+                "This DataFrame does not have a name and does not appear to be registered."
+            )
+
+        DataFrame.unregister_dataframe_by_name(self.name)
+
+        self.name = None  # Clear our internal DataFrame object name
+
+    def is_registered(self) -> bool:
+        """
+        Return True if the object is contained in the registry
+
+        Returns
+        -------
+        bool
+            Indicates if the object is contained in the registry
+
+        Raises
+        ------
+        RegistrationError
+            Raised if there's a server-side error or a mismatch of registered components
+
+        See Also
+        --------
+        register, attach, unregister, unregister_dataframe_by_name
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
+        """
+
+        if self.name is None:
+            return False
+
+        # Total number of registered parts should equal number of columns plus an entry
+        # for the index and the column order
+        total = len(self.data) + 2
+        registered = 0
+
+        for col, data in self.data.items():
+            if data.is_registered():
+                registered += 1
+
+        if self.index.values.is_registered():
+            registered += 1
+        if f"df_columns_{self.name}" in list_registry():
+            registered += 1
+
+        if 0 < registered < total:
+            warn(
+                f"WARNING: DataFrame {self.name} expected {total} components to be registered,"
+                f" but only located {registered}."
+            )
+            return False
+        else:
+            return registered == total
```

Comment:
we can just return `registered == total`

Reply:
Updated